kategory/src/main/kotlin/kategory/typeclasses/MonadFilter.kt

```diff
@@ -0,0 +1,17 @@
+package kategory
+
+/**
+ * a Monad equipped with an additional method which allows us to create an "empty" value for the Monad
+ * (for whatever "empty" makes sense for that particular monad). This is of particular interest to us
+ * since it allows us to add a filter method to a Monad, which is used when pattern matching or
+ * using guards in for comprehensions.
```

Comment:
:fire: the comment is only relevant to Scala.

Reply:
What do you mean? Yes, it's copied from Scala, but It's completely generic comment (not related to scala implementation) that applies equally here.

Reply:
The codebase has several comments that are only scala related:
https://github.com/kategory/kategory/blob/master/kategory/src/main/kotlin/kategory/data/Option.kt#L86
https://github.com/kategory/kategory/blob/master/kategory/src/main/kotlin/kategory/data/Option.kt#L181
https://github.com/kategory/kategory/blob/master/kategory/src/main/kotlin/kategory/data/Either.kt#L160

You can see subtle mistakes in all those links of examples that are not valid Kotlin and there is many more.
In this case there are terms used that are not called that in Kotlin:
- `for comprehensions` : We don't have `for` and therefore are either monadic comprehensions or we come up with new names but `for comprehensions` is a Scala thing.
- `pattern matching`: I don't think Kotlin has true pattern matching as it is in other langs like Haskell or Scala and even if we adopt this term in Kotlin we have no ocurrences of such an instance of using filter in when statements anywhere AFAIK.

My point is that since our implementation is deviating from the Scala one so much lately we may as well come up with our own comments or not provide them at all and leave it for docs so we use the terminology used and familiar to the Kotlin community.